x-pack/plugins/alerting/README.md

```diff
@@ -268,6 +269,61 @@ server.newPlatform.setup.plugins.alerting.registerType({
 });
 ```
 
+## Alert Navigation
+When registering an Alert Type, you'll likely want to provide a way of viewing alerts of that type within your own plugin, or perhaps you want to provide a view for all alerts created from within your solution within your own UI.
+
+In order for the Alerting framework to know that your plugin has its own internal view for displaying an alert, you must resigter a navigation handler within the framework.
+
+A navigation handler is nothing more than a function that receives an Alert and its corresponding AlertType, and is expected to then return the path *within your plugin* which knows how to display this alert.
+
+The signature of such a handler is:
+
+```
+type AlertNavigationHandler = (
```

Comment:
the function result type seems to be `JsonObject | string` in `alerting/public/alert_navigation_registry/types.ts`.  Are we not supporting `JsonObject` for now, at this level of the API?  Guessing we can support it, would need more tests and doc, etc, deferred to later ...

Reply:
This is for the State feature that doesn't work because we're booted by the management plugin which is legacy.
Once this is migrated this feature will work and only then can we add a functional test that would actually pass, which is why I don't mention it in the docs.

But once it's migrated navigation will work out of the box, without us having to do any more code for it - so it seemed worth keeping in.